src/guesstype.jl

```diff
@@ -26,14 +26,14 @@ function guessdateformat(str)
 
     for (typ, df) in dts
         x, len = try
-            tryparsenext_internal(typ, str, 1, endof(str), df)
+            tryparsenext_internal(typ, str, 1, lastindex(str), df)
         catch err
             continue
         end
-        if !isnull(x)
+        if x != nothing
```

Comment:
`!==` is typically better style when comparing w/ potential `nothing` values. I think it's more compiler friendly as well.